
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Dinindu Senanayake" name="author"/>
<link href="https://dinindusenanayake.github.io/wf-human-variation_split/" rel="canonical"/>
<link href="theme_images/NeSI_Logo-Icon_white.png" rel="icon"/>
<meta content="mkdocs-1.4.2, mkdocs-material-9.0.11" name="generator"/>
<title>wf-human_variation_split</title>
<link href="assets/stylesheets/main.0d440cfe.min.css" rel="stylesheet"/>
<link href="assets/stylesheets/palette.2505c338.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="" data-md-color-primary="green" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#epi2me-labswf-human-variation-for-nesi-mahuika-cluster">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="wf-human_variation_split" class="md-header__button md-logo" data-md-component="logo" href="." title="wf-human_variation_split">
<img alt="logo" src="theme_images/NeSI_Logo-Icon_white.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            wf-human_variation_split
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
<kbd>epi2me-labs/wf-human-variation</kbd> for NeSI Mahuika cluster
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="green" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="green" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="green" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/DininduSenanayake/wf-human-variation_split" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    DininduSenanayake/wf-human-variation_split
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-tabs__inner md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href=".">
<kbd>epi2me-labs/wf-human-variation</kbd> for NeSI Mahuika cluster
    </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="wf-human_variation_split" class="md-nav__button md-logo" data-md-component="logo" href="." title="wf-human_variation_split">
<img alt="logo" src="theme_images/NeSI_Logo-Icon_white.png"/>
</a>
    wf-human_variation_split
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/DininduSenanayake/wf-human-variation_split" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    DininduSenanayake/wf-human-variation_split
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<a class="md-nav__link md-nav__link--active" href=".">
<kbd>epi2me-labs/wf-human-variation</kbd> for NeSI Mahuika cluster
      </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="epi2me-labswf-human-variation-for-nesi-mahuika-cluster"><kbd>epi2me-labs/wf-human-variation</kbd> for NeSI Mahuika cluster<a class="headerlink" href="#epi2me-labswf-human-variation-for-nesi-mahuika-cluster" title="Permanent link">Â¶</a></h1>
<p>modified version of <kbd>epi2me-labs/wf-human-variation</kbd>        (<a href="https://github.com/epi2me-labs/wf-human-variation">https://github.com/epi2me-labs/wf-human-variation</a>) to make the workflow compatible with NeSI Mahuika cluster <kbd>Slurm scheduler</kbd></p>
<details class="terminal">
<summary>02-ont-guppy-gpu</summary>
<ul>
<li>Refer to NeSI Support documentation <a href="https://support.nesi.org.nz/hc/en-gb/articles/4546820344079-ont-guppy-gpu">https://support.nesi.org.nz/hc/en-gb/articles/4546820344079-ont-guppy-gpu</a></li>
</ul>
<p><kbd>Dorado</kbd></p>
<ul>
<li>Dorado is available as a module but treat as a R&amp;D application for the moment.</li>
<li>NeSI support documentation for Dorado <a href="https://support.nesi.org.nz/hc/en-gb/articles/6623692647951-Dorado">https://support.nesi.org.nz/hc/en-gb/articles/6623692647951-Dorado</a> </li>
</ul>
</details>
<details class="terminal">
<summary>03-ont-bam-merge.slurm</summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>!#/bin/bash -e
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>#SBATCH --account         nesi12345
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>#SBATCH --job-name        ont-bam-merge
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>#SBATCH --cpus-per-task   16
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a>#SBATCH --mem             64G
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a>#SBATCH --time            24:00:00
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a>module purge
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a>module load BamTools/2.5.2-GCC-11.3.0
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a>module load Sambamba/0.8.0
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a>
<a href="#__codelineno-0-13" id="__codelineno-0-13" name="__codelineno-0-13"></a># define variables
<a href="#__codelineno-0-14" id="__codelineno-0-14" name="__codelineno-0-14"></a>WKDIR='/path/to/wkir'                           #For an example : "/nesi/nobackup/nesi12345/.."
<a href="#__codelineno-0-15" id="__codelineno-0-15" name="__codelineno-0-15"></a>SAMPLE='PBXP289487'
<a href="#__codelineno-0-16" id="__codelineno-0-16" name="__codelineno-0-16"></a>
<a href="#__codelineno-0-17" id="__codelineno-0-17" name="__codelineno-0-17"></a># move to working dir
<a href="#__codelineno-0-18" id="__codelineno-0-18" name="__codelineno-0-18"></a>cd ${WKDIR}/${SAMPLE}/bam
<a href="#__codelineno-0-19" id="__codelineno-0-19" name="__codelineno-0-19"></a>
<a href="#__codelineno-0-20" id="__codelineno-0-20" name="__codelineno-0-20"></a># sambamba
<a href="#__codelineno-0-21" id="__codelineno-0-21" name="__codelineno-0-21"></a># sambamba merge -t 4 -p ${SAMPLE}_sorted_merged.bam pass/*.bam
<a href="#__codelineno-0-22" id="__codelineno-0-22" name="__codelineno-0-22"></a># bamtools
<a href="#__codelineno-0-23" id="__codelineno-0-23" name="__codelineno-0-23"></a>ls pass/*.bam &gt; bam_list.txt
<a href="#__codelineno-0-24" id="__codelineno-0-24" name="__codelineno-0-24"></a>bamtools merge -l bam_list.txt -out ${SAMPLE}_merged.bam
<a href="#__codelineno-0-25" id="__codelineno-0-25" name="__codelineno-0-25"></a>sambamba sort -m 52GB -t 16 ${SAMPLE}_merged.bam -o ${SAMPLE}_sorted_merged.bam
<a href="#__codelineno-0-26" id="__codelineno-0-26" name="__codelineno-0-26"></a>sambamba index -t 16 ${SAMPLE}_sorted_merged.bam
</code></pre></div>
</details>
<details class="terminal">
<summary>04-ont--human-variation-calling.slurm</summary>
<ul>
<li>
<p>Create a file named <em>nextflow_local_override.config</em> with the following content </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>executor {
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>$local {
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>           cpus = 48
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a>           memory = "128 GB"
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>       }
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a>         }
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a>
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a>singularity {
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a>  runOptions = "-B ${TMPDIR}"
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a>            }
</code></pre></div>
</li>
<li>
<p>Slurm script </p>
</li>
</ul>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>#!/bin/bash -e
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>#SBATCH --name          ont-human-variation
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a>#SBATCH --cpus-per-task 48
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>#SBATCH --mem           128G
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>#SBATCH --partition     milan
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a>#SBATCH --time          24:00:00
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a>
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a>module purge
<a href="#__codelineno-2-9" id="__codelineno-2-9" name="__codelineno-2-9"></a>module load Nextflow/22.10.3
<a href="#__codelineno-2-10" id="__codelineno-2-10" name="__codelineno-2-10"></a>module load Singularity/3.10.3
<a href="#__codelineno-2-11" id="__codelineno-2-11" name="__codelineno-2-11"></a>
<a href="#__codelineno-2-12" id="__codelineno-2-12" name="__codelineno-2-12"></a>
<a href="#__codelineno-2-13" id="__codelineno-2-13" name="__codelineno-2-13"></a># define variables
<a href="#__codelineno-2-14" id="__codelineno-2-14" name="__codelineno-2-14"></a>WKDIR='/path/to/workingdir'                           #For an example : "/nesi/nobackup/nesi12345/.."
<a href="#__codelineno-2-15" id="__codelineno-2-15" name="__codelineno-2-15"></a>SAMPLE='PBXP289487'
<a href="#__codelineno-2-16" id="__codelineno-2-16" name="__codelineno-2-16"></a>MODEL='/path/to/clair3_models/ont_guppy5'
<a href="#__codelineno-2-17" id="__codelineno-2-17" name="__codelineno-2-17"></a>REFERENCE='/path/to/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta'
<a href="#__codelineno-2-18" id="__codelineno-2-18" name="__codelineno-2-18"></a>TDREPETS='/path/to/bed/reference/human_GRCh38_no_alt_analysis_set.trf.bed'
<a href="#__codelineno-2-19" id="__codelineno-2-19" name="__codelineno-2-19"></a>OUTPUT='results'
<a href="#__codelineno-2-20" id="__codelineno-2-20" name="__codelineno-2-20"></a>
<a href="#__codelineno-2-21" id="__codelineno-2-21" name="__codelineno-2-21"></a>#Singularity and Nextflow variables
<a href="#__codelineno-2-22" id="__codelineno-2-22" name="__codelineno-2-22"></a>export SINGULARITY_TMPDIR=/some/path/in/nobackup/singularity_cache
<a href="#__codelineno-2-23" id="__codelineno-2-23" name="__codelineno-2-23"></a>export SINGULARITY_CACHEDIR=$SINGULARITY_TMPDIR
<a href="#__codelineno-2-24" id="__codelineno-2-24" name="__codelineno-2-24"></a>setfacl -b "$SINGULARITY_TMPDIR"
<a href="#__codelineno-2-25" id="__codelineno-2-25" name="__codelineno-2-25"></a>
<a href="#__codelineno-2-26" id="__codelineno-2-26" name="__codelineno-2-26"></a>export NXF_EXECUTOR=slurm
<a href="#__codelineno-2-27" id="__codelineno-2-27" name="__codelineno-2-27"></a>export NXF_SINGULARITY_CACHEDIR=$SINGULARITY_CACHEDIR
<a href="#__codelineno-2-28" id="__codelineno-2-28" name="__codelineno-2-28"></a>
<a href="#__codelineno-2-29" id="__codelineno-2-29" name="__codelineno-2-29"></a>NFCONFIG='/path/to/nextflow_local_overide.config'
<a href="#__codelineno-2-30" id="__codelineno-2-30" name="__codelineno-2-30"></a># note: created an overide config to provide modified CPU and Memory values
<a href="#__codelineno-2-31" id="__codelineno-2-31" name="__codelineno-2-31"></a># change these values if you want to tweak performance based on resources
<a href="#__codelineno-2-32" id="__codelineno-2-32" name="__codelineno-2-32"></a>
<a href="#__codelineno-2-33" id="__codelineno-2-33" name="__codelineno-2-33"></a># move to working dir
<a href="#__codelineno-2-34" id="__codelineno-2-34" name="__codelineno-2-34"></a>cd ${WKDIR}/${SAMPLE}
<a href="#__codelineno-2-35" id="__codelineno-2-35" name="__codelineno-2-35"></a>
<a href="#__codelineno-2-36" id="__codelineno-2-36" name="__codelineno-2-36"></a># pull nextflow pipeline (if haven't already)
<a href="#__codelineno-2-37" id="__codelineno-2-37" name="__codelineno-2-37"></a>nextflow run epi2me-labs/wf-human-variation --help
<a href="#__codelineno-2-38" id="__codelineno-2-38" name="__codelineno-2-38"></a>
<a href="#__codelineno-2-39" id="__codelineno-2-39" name="__codelineno-2-39"></a># run Clair3 variant calling
<a href="#__codelineno-2-40" id="__codelineno-2-40" name="__codelineno-2-40"></a>nextflow -c ${NFCONFIG} run epi2me-labs/wf-human-variation \
<a href="#__codelineno-2-41" id="__codelineno-2-41" name="__codelineno-2-41"></a>  -resume \
<a href="#__codelineno-2-42" id="__codelineno-2-42" name="__codelineno-2-42"></a>  --threads ${SLURM_CPUS_PER_TASK} \
<a href="#__codelineno-2-43" id="__codelineno-2-43" name="__codelineno-2-43"></a>  -profile singularity,local \
<a href="#__codelineno-2-44" id="__codelineno-2-44" name="__codelineno-2-44"></a>  --snp --sv \
<a href="#__codelineno-2-45" id="__codelineno-2-45" name="__codelineno-2-45"></a>  --phase_vcf \
<a href="#__codelineno-2-46" id="__codelineno-2-46" name="__codelineno-2-46"></a>  --use_longphase \
<a href="#__codelineno-2-47" id="__codelineno-2-47" name="__codelineno-2-47"></a>  --tr_bed ${TDREPETS} \
<a href="#__codelineno-2-48" id="__codelineno-2-48" name="__codelineno-2-48"></a>  --model ${MODEL} \
<a href="#__codelineno-2-49" id="__codelineno-2-49" name="__codelineno-2-49"></a>  --bam ./bam/${SAMPLE}_sorted_merged.bam \
<a href="#__codelineno-2-50" id="__codelineno-2-50" name="__codelineno-2-50"></a>  --ref ${REFERENCE} \
<a href="#__codelineno-2-51" id="__codelineno-2-51" name="__codelineno-2-51"></a>  --out_dir ${OUTPUT}
</code></pre></div>
</details>
<details class="terminal">
<summary>05-ont-whatshap-phase.slurm</summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>#!/bin/bash -e
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>#SBATCH --account         nesi12345
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>#SBATCH --job-name        ont-whatshap-phase
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>#SBATCH --cpus-per-task   16
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a>#SBATCH --mem             64G
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a>#SBATCH --time            24:00:00
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a>
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a>module purge
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a>module load WhatsHap/1.6-gimkl-2022a-Python-3.10.5
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>module load SAMtools/1.16.1-GCC-11.3.0
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a>
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a># define variables
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a>WKDIR='/path/to/workingdir'
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a>SAMPLE='PBXP289487'
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a>REFERENCE='/path/to/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta'
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a>OUTPUT='results'
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a>
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a># move to working dir
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a>cd ${WKDIR}/${SAMPLE}
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a>
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a># whatshap phase tagging of bam output
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a>whatshap haplotag \
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a>    --ignore-read-groups \
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>    --output ./bam/${SAMPLE}_sorted_merged.hp.bam  \
<a href="#__codelineno-3-26" id="__codelineno-3-26" name="__codelineno-3-26"></a>    --reference ${REFERENCE} \
<a href="#__codelineno-3-27" id="__codelineno-3-27" name="__codelineno-3-27"></a>    ${OUTPUT}/all.wf_snp.vcf.gz ./bam/${SAMPLE}_sorted_merged.bam
<a href="#__codelineno-3-28" id="__codelineno-3-28" name="__codelineno-3-28"></a># index bam
<a href="#__codelineno-3-29" id="__codelineno-3-29" name="__codelineno-3-29"></a>samtools index -@ 16 ./bam/${SAMPLE}_sorted_merged.hp.bam
<a href="#__codelineno-3-30" id="__codelineno-3-30" name="__codelineno-3-30"></a>
<a href="#__codelineno-3-31" id="__codelineno-3-31" name="__codelineno-3-31"></a># Notes:
<a href="#__codelineno-3-32" id="__codelineno-3-32" name="__codelineno-3-32"></a># this step phases the data based on the clair3 output and generates a 
<a href="#__codelineno-3-33" id="__codelineno-3-33" name="__codelineno-3-33"></a># phased bam file. This contains information assigning reads to each
<a href="#__codelineno-3-34" id="__codelineno-3-34" name="__codelineno-3-34"></a># haplotype. At this stage the bam is able to loaded into a genome 
<a href="#__codelineno-3-35" id="__codelineno-3-35" name="__codelineno-3-35"></a># viewer/browser and contains aligned reads, with base modification
<a href="#__codelineno-3-36" id="__codelineno-3-36" name="__codelineno-3-36"></a># (methylation) information, as well as the haplotype information.
</code></pre></div>
</details>
<details class="terminal">
<summary>06-ont-methyl-calling.slurm</summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>#!/bin/bash -e
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>#SBATCH --account         nesi12345
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a>#SBATCH --job-name        ont-bam-merge
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a>#SBATCH --cpus-per-task   16
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a>#SBATCH --mem             32G
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a>#SBATCH --time            08:00:00
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a>
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a>module purge
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a>module load modbam2bed/0.9.4-GCC-11.3.0
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a>
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a># define variables
<a href="#__codelineno-4-13" id="__codelineno-4-13" name="__codelineno-4-13"></a>WKDIR='/path/to/workdir'
<a href="#__codelineno-4-14" id="__codelineno-4-14" name="__codelineno-4-14"></a>SAMPLE='PBXP289487'
<a href="#__codelineno-4-15" id="__codelineno-4-15" name="__codelineno-4-15"></a>REFERENCE='/path/to/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta'
<a href="#__codelineno-4-16" id="__codelineno-4-16" name="__codelineno-4-16"></a>
<a href="#__codelineno-4-17" id="__codelineno-4-17" name="__codelineno-4-17"></a># move to working dir
<a href="#__codelineno-4-18" id="__codelineno-4-18" name="__codelineno-4-18"></a>cd ${WKDIR}/${SAMPLE}
<a href="#__codelineno-4-19" id="__codelineno-4-19" name="__codelineno-4-19"></a>
<a href="#__codelineno-4-20" id="__codelineno-4-20" name="__codelineno-4-20"></a># create methylation bed files
<a href="#__codelineno-4-21" id="__codelineno-4-21" name="__codelineno-4-21"></a>for HP in 1 2; do
<a href="#__codelineno-4-22" id="__codelineno-4-22" name="__codelineno-4-22"></a>    modbam2bed \
<a href="#__codelineno-4-23" id="__codelineno-4-23" name="__codelineno-4-23"></a>        -e -m 5mC --cpg -t 16 --haplotype ${HP} \
<a href="#__codelineno-4-24" id="__codelineno-4-24" name="__codelineno-4-24"></a>        ${REFERENCE} \
<a href="#__codelineno-4-25" id="__codelineno-4-25" name="__codelineno-4-25"></a>        ./bam/${SAMPLE}_sorted_merged.hp.bam \
<a href="#__codelineno-4-26" id="__codelineno-4-26" name="__codelineno-4-26"></a>        | bgzip -c &gt; ./bed/${SAMPLE}_methylation.hp${HP}.cpg.bed.gz
<a href="#__codelineno-4-27" id="__codelineno-4-27" name="__codelineno-4-27"></a>done;
<a href="#__codelineno-4-28" id="__codelineno-4-28" name="__codelineno-4-28"></a>
<a href="#__codelineno-4-29" id="__codelineno-4-29" name="__codelineno-4-29"></a># create an aggregated bed file
<a href="#__codelineno-4-30" id="__codelineno-4-30" name="__codelineno-4-30"></a>modbam2bed \
<a href="#__codelineno-4-31" id="__codelineno-4-31" name="__codelineno-4-31"></a>  -e -m 5mC --cpg --aggregate -t 16 \
<a href="#__codelineno-4-32" id="__codelineno-4-32" name="__codelineno-4-32"></a>  ${REFERENCE} \
<a href="#__codelineno-4-33" id="__codelineno-4-33" name="__codelineno-4-33"></a>  ./bam/${SAMPLE}_sorted_merged.hp.bam \
<a href="#__codelineno-4-34" id="__codelineno-4-34" name="__codelineno-4-34"></a>  | bgzip -c &gt; ./bed/${SAMPLE}_methylation.aggregated.cpg.bed.gz
<a href="#__codelineno-4-35" id="__codelineno-4-35" name="__codelineno-4-35"></a>
<a href="#__codelineno-4-36" id="__codelineno-4-36" name="__codelineno-4-36"></a># Notes:
<a href="#__codelineno-4-37" id="__codelineno-4-37" name="__codelineno-4-37"></a># this step extracts the methylation information from the bam file, generating
<a href="#__codelineno-4-38" id="__codelineno-4-38" name="__codelineno-4-38"></a># bed files. There are two processes here, the first creates two bed files, one
<a href="#__codelineno-4-39" id="__codelineno-4-39" name="__codelineno-4-39"></a># for each haplotype. These are very useful for exploring allele specific methylation.
<a href="#__codelineno-4-40" id="__codelineno-4-40" name="__codelineno-4-40"></a># The second process generates a single aggragated bed file, it merges the
<a href="#__codelineno-4-41" id="__codelineno-4-41" name="__codelineno-4-41"></a># haplotype data to give site specific methylation. It should be noted that this is
<a href="#__codelineno-4-42" id="__codelineno-4-42" name="__codelineno-4-42"></a># per strand, so some processing will be required to 'collapse' the data to a
<a href="#__codelineno-4-43" id="__codelineno-4-43" name="__codelineno-4-43"></a># single CpG site, but this type of work is usually performed in the downstream\
<a href="#__codelineno-4-44" id="__codelineno-4-44" name="__codelineno-4-44"></a># analysis, using packages such as methylkit.
</code></pre></div>
</details>
<details class="terminal">
<summary>07-ont-sv-cutesv.slurm</summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>#!/bin/bash -e
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>#SBATCH --account         nesi12345
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>#SBATCH --job-name        ont-csv-cutesv
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>#SBATCH --cpus-per-task   16
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a>#SBATCH --mem             128G
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a>#SBATCH --time            06:00:00
<a href="#__codelineno-5-8" id="__codelineno-5-8" name="__codelineno-5-8"></a>
<a href="#__codelineno-5-9" id="__codelineno-5-9" name="__codelineno-5-9"></a>module purge
<a href="#__codelineno-5-10" id="__codelineno-5-10" name="__codelineno-5-10"></a>module load cuteSV/2.0.2-gimkl-2020a-Python-3.8.2
<a href="#__codelineno-5-11" id="__codelineno-5-11" name="__codelineno-5-11"></a>
<a href="#__codelineno-5-12" id="__codelineno-5-12" name="__codelineno-5-12"></a># define variables
<a href="#__codelineno-5-13" id="__codelineno-5-13" name="__codelineno-5-13"></a>WKDIR='/path/to/workingdir'
<a href="#__codelineno-5-14" id="__codelineno-5-14" name="__codelineno-5-14"></a>SAMPLE='PBXP289487'
<a href="#__codelineno-5-15" id="__codelineno-5-15" name="__codelineno-5-15"></a>REFERENCE='/path/to/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta'
<a href="#__codelineno-5-16" id="__codelineno-5-16" name="__codelineno-5-16"></a>OUTPUT='results'
<a href="#__codelineno-5-17" id="__codelineno-5-17" name="__codelineno-5-17"></a>
<a href="#__codelineno-5-18" id="__codelineno-5-18" name="__codelineno-5-18"></a># move to working dir
<a href="#__codelineno-5-19" id="__codelineno-5-19" name="__codelineno-5-19"></a>cd ${WKDIR}/${SAMPLE}
<a href="#__codelineno-5-20" id="__codelineno-5-20" name="__codelineno-5-20"></a>mkdir cutesv_tmp
<a href="#__codelineno-5-21" id="__codelineno-5-21" name="__codelineno-5-21"></a>
<a href="#__codelineno-5-22" id="__codelineno-5-22" name="__codelineno-5-22"></a># cuteSV processing
<a href="#__codelineno-5-23" id="__codelineno-5-23" name="__codelineno-5-23"></a>cuteSV -t ${SLURM_CPUS_PER_TASK} \
<a href="#__codelineno-5-24" id="__codelineno-5-24" name="__codelineno-5-24"></a>  --max_cluster_bias_INS 100 \
<a href="#__codelineno-5-25" id="__codelineno-5-25" name="__codelineno-5-25"></a>  --diff_ratio_merging_INS 0.3 \
<a href="#__codelineno-5-26" id="__codelineno-5-26" name="__codelineno-5-26"></a>  --max_cluster_bias_DEL 100 \
<a href="#__codelineno-5-27" id="__codelineno-5-27" name="__codelineno-5-27"></a>  --diff_ratio_merging_DEL 0.3 \
<a href="#__codelineno-5-28" id="__codelineno-5-28" name="__codelineno-5-28"></a>  ./bam/${SAMPLE}_sorted_merged.hp.bam \
<a href="#__codelineno-5-29" id="__codelineno-5-29" name="__codelineno-5-29"></a>  ${REFERENCE} \
<a href="#__codelineno-5-30" id="__codelineno-5-30" name="__codelineno-5-30"></a>  ${OUTPUT}/${SAMPLE}_sv_cutesv.vcf \
<a href="#__codelineno-5-31" id="__codelineno-5-31" name="__codelineno-5-31"></a>  ./cutesv_tmp
<a href="#__codelineno-5-32" id="__codelineno-5-32" name="__codelineno-5-32"></a>
<a href="#__codelineno-5-33" id="__codelineno-5-33" name="__codelineno-5-33"></a># Notes:
<a href="#__codelineno-5-34" id="__codelineno-5-34" name="__codelineno-5-34"></a># this step is for evaluation of another structural variant caller, cuteSV. It's 
<a href="#__codelineno-5-35" id="__codelineno-5-35" name="__codelineno-5-35"></a># often nice to have the ability to compare results between various tools. As 
<a href="#__codelineno-5-36" id="__codelineno-5-36" name="__codelineno-5-36"></a># SVs are important to this project this step has been included in the process.
<a href="#__codelineno-5-37" id="__codelineno-5-37" name="__codelineno-5-37"></a># For other projects it may well be enough to stop after processing step 06.
<a href="#__codelineno-5-38" id="__codelineno-5-38" name="__codelineno-5-38"></a># This process outputs a vcf file with the structural variation recorded per
<a href="#__codelineno-5-39" id="__codelineno-5-39" name="__codelineno-5-39"></a># line.
</code></pre></div>
</details>
</article>
</div>
</div>
<a class="md-top md-icon" data-md-component="top" hidden="" href="#">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </a>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": ".", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="assets/javascripts/bundle.6df46069.min.js"></script>
<script src="javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>